addons/io_hubs_addon/components/operators.py

```diff
@@ -433,6 +442,79 @@ def invoke(self, context, event):
         return wm.invoke_props_dialog(self, width=600)
 
 
+class GenerateNavMesh(Operator):
+    bl_idname = "object.generate_hubs_navmesh"
+    bl_label = "Generate Hubs Navigation Mesh"
+    bl_options = {'REGISTER', 'UNDO'}
+
+    def execute(self, context):
+        import bmesh
+
+        obj = bpy.context.active_object
+
+        # Make sure there's a mesh object set as active. If not, abort
+        if obj is not None:
+            if obj.type != "MESH":
+                return {'FINISHED'}
+        else:
+            return {'FINISHED'}
+
+        bm = bmesh.new()
+        bm.from_mesh(obj.data)
```

Comment:
You probably need to apply transforms here so the actual geometry is transformed. Something like:
```python
        bpy.ops.object.transform_apply(location=False, rotation=True, scale=True)
```

Reply:
Maybe also applying operators could be a good thing to do as the user probably expects the nav-mesh generation to happen on the resulting mesh. nav-mesh generation for surfaces?